ui/src/app/pages/data/data-set/data-set-component.tsx

```diff
@@ -1325,38 +1336,47 @@ export const DataSetComponent = fp.flow(withUserProfile(), withCurrentWorkspace(
                             content='Requires Owner or Writer permission' disabled={this.canWrite}>
               {this.isCreatingNewDataset ?
                 <Button style={{marginBottom: '2rem', marginRight: '1rem'}} data-test-id='save-button'
-                        onClick ={() => this.setState({openCreateModal: true})}
+                        onClick ={() => this.setState({modalState: ModalState.Create})}
                         disabled={this.disableSave() || !this.canWrite || !dataSetTouched}>
                   Create Dataset
                 </Button> :
                 <Button style={{marginBottom: '2rem', marginRight: '1rem'}} data-test-id='save-button'
                         onClick ={() => this.saveDataset()}
                         disabled={this.state.savingDataset || this.disableSave() || !this.canWrite || !dataSetTouched}>
-                  Save
+                  Save Dataset
                 </Button>}
             </TooltipTrigger>
 
           <TooltipTrigger data-test-id='export-tooltip'
                           content={exportError}
                           disabled={!exportError}>
             <Button style={{marginBottom: '2rem'}} data-test-id='save-button'
-                    onClick ={() => this.setState({openExportModal: true})}
+                    onClick ={() => this.onClickExport()}
                     disabled={this.disableSave() || !!exportError}>
-              Export
+              Analyze
             </Button>
           </TooltipTrigger>
         </div>
-        {this.state.openCreateModal && <CreateModal entityName='Dataset'
+        {this.state.modalState === ModalState.Create && <CreateModal entityName='Dataset'
                                                     getExistingNames={async() => {
                                                       const resources = await workspacesApi().getWorkspaceResources(namespace, id,
                                                         {typesToFetch: [ResourceType.DATASET]});
                                                       return resources.map(resource => resource.dataSet.name);
                                                     }}
                                                     save={(name, desc) => this.createDataset(name, desc)}
-                                                    close={() => this.setState({openCreateModal: false})}/>}
-
-        {openExportModal && <ExportDatasetModal dataset={dataSet}
-                                                closeFunction={() => this.setState({openExportModal: false})}/>}
+                                                    close={() => this.setState({modalState: ModalState.None})}/>}
+
+        {this.state.modalState === ModalState.Export && <ExportDatasetModal dataset={dataSet}
```

Comment:
Is there any coverage to add in `dataset-page.spec.tsx`? No idea why the test file is named so differently from this..

Reply:
sure - figuring out the right things to mock was a bit tricky but I added some cases

Reply:
also, I renamed the component to match the test file.